### DISCUSSION TITLE
How to get speed of Srcollbar

### DISCUSSION THREAD

**Discussion by cpiocova** [idiotWu/smooth-scrollbar#344] [Q&A] [answered]
Hello thank you for this awesome plugin!

Im kinda new in this and i have a question.

I would like to get the speed data from the scroll, like you guys have in your page (https://idiotwu.github.io/smooth-scrollbar/) when you user a plot to show the speed curve.

Is there any function that i should be calling to get the value of the speed scroll.

For example i would like to get the offset doing an:

**scrollbar.addListener( () => {scrollbar.offset})**

And then, get the offset for X and Y.

Is there any way to get the speed like:
 **scrollbar.addListener( () => {scrollbar.speed})**

Because i would like to animate dom elements using the speed that i get for every addListener value i get

Kind Regards,

Pio

**Comment by dolphin-wood** [comment, marked as answer]
Since the speed is given by

![equation](https://latex.codecogs.com/svg.latex?%5Cinline%20speed%20%3D%20%5Cleft%7C%20%5Cfrac%7B%5CDelta%20offset%7D%7B%5CDelta%20t%7D%20%5Cright%7C)

you need to record the last offsets and compare it with current ones.

Here is the implementation from the demo page: https://github.com/idiotWu/smooth-scrollbar/blob/b8e4005e789c168bce446e8479c1186d6ee59d0e/__demo__/scripts/monitor.ts#L414-L435

**Comment by cpiocova** [comment]
Thank's my friends! Solved!!
